test/shopify_api/graphql/response_test.exs

```diff
@@ -0,0 +1,181 @@
+defmodule ShopifyAPI.GraphQL.ResponseTest do
+  use ExUnit.Case
+
+  alias ShopifyAPI.GraphQL.Response
+
+  @httpoison_response {:ok,
+                       %HTTPoison.Response{
+                         body:
+                           "{\"data\":{\"metafield1\":{\"deletedId\":\"gid://shopify/Metafield/5256098316335\",\"userErrors\":[]}},\"extensions\":{\"cost\":{\"actualQueryCost\":10,\"fields\":[{\"definedCost\":10,\"path\":[\"metafield1\"],\"requestedChildrenCost\":0,\"requestedTotalCost\":10}],\"requestedQueryCost\":10,\"throttleStatus\":{\"currentlyAvailable\":990,\"maximumAvailable\":1.0e3,\"restoreRate\":50.0}}}}",
+                         headers: [
+                           {"cache-control", "max-age=0, private, must-revalidate"},
+                           {"content-length", "352"},
+                           {"date", "Tue, 20 Aug 2019 20:27:42 GMT"},
+                           {"server", "Cowboy"}
+                         ],
+                         request: %HTTPoison.Request{
+                           body:
+                             "mutation {\n    metafield1: metafieldDelete (input: {id: \"gid://shopify/Metafield/123456789\"}){\n    deletedId\n    userErrors {\n      field\n      message\n      }\n    }\n  }",
+                           headers: [
+                             {"Content-Type", "application/graphql"},
+                             {"X-Shopify-Access-Token", "1234"}
+                           ],
+                           method: :post,
+                           options: [
+                             token: %ShopifyAPI.AuthToken{
+                               app_name: "",
+                               code: "",
+                               plus: false,
+                               shop_name: "localhost:60370",
+                               timestamp: 0,
+                               token: "1234"
+                             }
+                           ],
+                           params: %{},
+                           url: "http://localhost:60370/admin/api/2019-07/graphql.json"
+                         },
+                         request_url: "http://localhost:60370/admin/api/2019-07/graphql.json",
+                         status_code: 200
+                       }}
```

Comment:
It's not in scope for this PR, but I think we'll need to start considering more carefully how we work with and store API response fixtures in tests sometime soon.